bluemira/geometry/tools.py

```diff
@@ -1868,6 +1868,94 @@ def raise_error_if_overlap(
         raise GeometryError(f"{origin_name} and {target_name} partially/fully overlaps!")
 
 
+# ======================================================================================
+# Boolean operations
+# ======================================================================================
+
+
+def make_compound(shapes: Iterable[BluemiraGeo], label: str = "") -> BluemiraGeo:
```

Comment:
these two functions are exactly the same apart from the one line make_compound/join_connect, can we reduce the repetition later?